scipy/spatial/transform/_rotation.pyx

```diff
@@ -1273,6 +1345,200 @@ cdef class Rotation:
+                             "sequences or 'i'/'intrinsic' for intrinsic "
+                             "sequences, got {}".format(order))
+
+        axes = np.asarray(axes)
+        if axes.ndim == 1:
+            axes = axes.reshape([1, 3])
+
+        num_axes = len(axes)
+
+        for axis in axes:
+            if len(axis) != 3:
+                raise ValueError("Axes must be vectors of length 3.")
+
+        if num_axes < 1 or num_axes > 3:
+            raise ValueError("Expected up to 3 axes, got {}".format(num_axes))
+
+        # normalize axes
+        norm = np.repeat(np.linalg.norm(axes, axis=1), 3)
+        axes = axes / norm.reshape(num_axes, 3)
+
+        if (num_axes > 1 and abs(np.dot(axes[0], axes[1])) >= 1e-7 or 
+            num_axes > 2 and abs(np.dot(axes[1], axes[2])) >= 1e-7):
+            raise ValueError("Consecutive axes must be orthogonal.")
+            
+        angles = np.asarray(angles, dtype=float)
+        if degrees:
+            angles = np.deg2rad(angles)
+
+        is_single = False
+        # Prepare angles to have shape (num_rot, num_axes)
+        if num_axes == 1:
+            if angles.ndim == 0:
+                # (1, 1)
+                angles = angles.reshape((1, 1))
+                is_single = True
+            elif angles.ndim == 1:
+                # (N, 1)
+                angles = angles[:, None]
+            elif angles.ndim == 2 and angles.shape[-1] != 1:
+                raise ValueError("Expected `angles` parameter to have shape "
+                                "(N, 1), got {}.".format(angles.shape))
+            elif angles.ndim > 2:
+                raise ValueError("Expected float, 1D array, or 2D array for "
+                                "parameter `angles` corresponding to `seq`, "
+                                "got shape {}.".format(angles.shape))
+        else:  # 2 or 3 axes
+            if angles.ndim not in [1, 2] or angles.shape[-1] != num_axes:
+                raise ValueError("Expected `angles` to be at most "
+                                "2-dimensional with width equal to number "
+                                "of axes specified, got {} for shape".format(
+                                angles.shape))
+
+            if angles.ndim == 1:
+                # (1, num_axes)
+                angles = angles[None, :]
+                is_single = True
+
+        # By now angles should have shape (num_rot, num_axes)
+        # sanity check
+        if angles.ndim != 2 or angles.shape[-1] != num_axes:
+            raise ValueError("Expected angles to have shape (num_rotations, "
+                            "num_axes), got {}.".format(angles.shape))
+
+        q = Rotation.identity(len(angles))
+        for i in range(num_axes):
+            qi = Rotation.from_rotvec(angles[:, i, np.newaxis] * axes[i])
+            if extrinsic:
+                q = qi * q
+            else:
+                q = q * qi
```

Comment:
This code doesn't look very Pythonic. Maybe something with `math.prod` , `reversed`, `zip` can be done. But it's definitely minor.

Reply:
Is it a good idea to import `math` only for this though, given that it's at most a multiplication of 3 terms?

Reply:
It's definitely not necessary.

Reply:
I tried both `numpy.prod` and `math.prod` but I think they are defined for numerical values only, if that's okay I'd leave it as is for now!

Reply:
I would think that `math.prod` should work with any type supporting `*`, but maybe I'm wrong. Anyway this won't bring anything significant.